pkg/services/mysqldb/provision.go

```diff
@@ -27,7 +29,47 @@ func (s *serviceManager) ValidateProvisioningParameters(
 		sslEnforcement != "disabled" {
 		return service.NewValidationError(
 			"sslEnforcement",
-			fmt.Sprintf(`invalid sslEnforcement option: "%s"`, pp.SSLEnforcement),
+			fmt.Sprintf(`invalid option: "%s"`, pp.SSLEnforcement),
+		)
+	}
+	if pp.FirewallIPStart != "" || pp.FirewallIPEnd != "" {
+		if pp.FirewallIPStart == "" {
+			return service.NewValidationError(
+				"firewallStartIPAddress",
+				fmt.Sprintf(`invalid option: "%s"`, pp.FirewallIPStart),
```

Comment:
Can we change the error text? Something that explains that when one of these fields is set, they both must be.

Same goes for the next check. Ditto for the other modules.